config/class-sitedetailsindex.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Automattic\VIP\Config;
+
+class SiteDetailsIndex {
+	private static $_instance;
+
+	private const LOG_FEATURE_NAME = 'site_details_index';
+
+	public static function instance() {
+		if ( ! ( static::$_instance instanceof SiteDetailsIndex ) ) {
+			static::$_instance = new SiteDetailsIndex();
+			static::$_instance->init();
+		}
+
+		return static::$_instance;
+	}
+
+	public function init() {
+		add_filter( 'vip_site_details_index_data', [ $this, 'set_env_and_core' ] );
+	}
+
+	/**
+	 * Given the site details from the vip_site_details_index_data filter, set the environment and core values
+	 *
+	 * @param {array} $site_details The current site details
+	 * @return {array} The new site details
+	 */
+	public function set_env_and_core( $site_details ) {
+		global $wp_version;
+
+		$site_id = 0;
+		if ( defined( 'VIP_GO_APP_ID' ) && VIP_GO_APP_ID ) {
+			$site_id = VIP_GO_APP_ID;
+		} else {
+			if ( defined( 'FILES_CLIENT_SITE_ID' ) && FILES_CLIENT_SITE_ID ) {
+				$site_id = FILES_CLIENT_SITE_ID;
+			}
+		}
+
+		// log2logstash has it's own timestamp field as well. However, if/when we ever decide
+		// to swap indexes, we'll need our own. Added here so that the migration will be easier
+		// and all the fields we need are included in the actual site details index data.
+		$site_details['timestamp'] = time();
+		$site_details['client_site_id'] = $site_id;
+		$site_details['environment_id'] = $site_id;
+		$site_details['environment_name'] = strval( get_bloginfo( 'name' ) );
```

Comment:
`get_bloginfo( 'name' )` returns the title for the site which is different from what we refer to as the "environment name" so this will get confusing. I think we probably want `VIP_GO_APP_ENVIRONMENT`?